Documentation Updates (#50)
oindrillac/ttmtool: README.md

This mode will initiate the model training process by following the steps of data collection, feature engineering, model training on the PR data available and finally running the inference i.e. predicting the time to merge for the latest PR on the repository. 

(*NOTE : This workflow will fail if there are no PRs on the repository*)

You can also initiate a new trigger by going to actions for your repository like [here](https://github.com/aicoe-aiops/ocp-ci-analysis/actions/workflows/train-ttm.yaml):

![image](https://user-images.githubusercontent.com/26301643/206544812-b6ffbe44-7bd3-4c7d-ab75-55b29d24f8f4.png)

Go select - `Run Time to Merge Model Training` and go to `Run workflow` on upper right and run it like such :

![image](https://user-images.githubusercontent.com/26301643/206548995-5e82cf06-1892-462d-9114-e81065b1f13e.png))

This will initiate the model training and inference action. 

> 2. **Infernce Mode** : 

Reviewer: ```suggestion
2. **Inference Mode** : 
```

Similar to the `train-ttm.yaml` file, you can add another file called `predict-ttm.yaml` file to `.github/worklows/` like [this](https://github.com/aicoe-aiops/ocp-ci-analysis/blob/master/.github/workflows/predict-ttm.yaml). This file has `MODE` as `0` which would enable just inference on the new incoming Pull Request and add a comment to the pull request specifying the approximate time it will take to be merged.

> 

Reviewer: Do you want to add a section here to say if you would like to try this tool out without adding the workflow file to your repository. And keep a short version of some of the instructions from the previous version. This might be useful fork forks who just want to try this out without setting this up in their repo. If that is making the readme too big, maybe you can pull that out into a different doc and point to the detailed instructions in the previous version of the readme from this section keeping this here short.

## Alternate method

You can fork this repository and add secrets your fork and run the training by using the bash script or by manually triggering the workflow....

Author: Sounds good, will add this too

This will initiate the model training and inference action. 

2. **Inference Mode** : 

Similar to the `train-ttm.yaml` file, you can add another file called `predict-ttm.yaml` file to `.github/worklows/` like [this](https://github.com/aicoe-aiops/ocp-ci-analysis/blob/master/.github/workflows/predict-ttm.yaml). This file has `MODE` as `0` which would enable just inference on the new incoming Pull Request and add a comment to the pull request specifying the approximate time it will take to be merged.

![image](https://user-images.githubusercontent.com/26301643/206541965-c85eb5f8-012e-454c-9f0d-467db0c8be07.png)

## Step 3

To view your running workflow from the Github UI, go to "Actions" and click on the workflow run :

![image](https://user-images.githubusercontent.com/26301643/206549380-369e6680-5e4e-4994-ae8a-59fd149db426.png)

Click on `pipeline` to see logs and errors :

![image](https://user-images.githubusercontent.com/26301643/206549621-0184880a-769e-48dc-928f-44c721464999.png)

## Architecture

![ttm github workflow (1)](https://user-images.githubusercontent.com/26301643/206780355-7d4e877c-c9ed-448d-8331-ad31e865b85f.png)

## Alternate Approach

> You can also use this tool for your repository using an alternate approach. Here are the steps to follow :

Reviewer: ```suggestion
You can also use train this model on your repository using an alternate approach without adding the workflow file to your repository. Here are the steps to follow :
```

1. Fork this [repository](https://github.com/redhat-et/time-to-merge-tool) and to your fork add the secrets as mentioned [here](https://github.com/redhat-et/time-to-merge-tool#step-1). Make sure to mention the `REPO` and `ORG` for the repository you want to run TTM on.

> 2. Go to `Actions` for your fork and select the workflow you want to run. You can also mention the mode of action by changing it [here](https://github.com/redhat-et/time-to-merge-tool/blob/016897ab03a9ff0aa800d6bcaececad028c93841/.github/workflows/inference.yaml#L22)

Reviewer: ```suggestion
2. Go to `Actions` for your fork and select the `run in container` workflow to train the model. 
```

> <img width="1056" alt="Screen Shot 2022-12-09 at 12 43 26 PM" src="https://user-images.githubusercontent.com/26301643/206761255-d74f83d9-183e-4ad9-97da-936f61db36a2.png">

Reviewer: can we change this image to point to the training workflow?

Author: Sure

> 3. You can also interact with this tool by POST request to Github API endpoint. From your terminal, clone your repository and run `bash run-ttm.sh` .

Reviewer: ```suggestion
3. You can also interact with this tool by POST request to Github API endpoint. From your terminal, clone your repository and run `bash run-ttm.sh` . This will run the training workflow and train the TTM model on the repo and org of your choice.
```
